heft-plugins/heft-storybook-plugin/src/schemas/heft-storybook-plugin.schema.json

```diff
@@ -0,0 +1,25 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "title": "Heft Storybook Plugin Options Configuration",
+  "description": "This schema describes the \"options\" field that can be specified in heft.json when loading \"@rushstack/heft-storybook-plugin\".",
+  "type": "object",
+  "additionalProperties": false,
+
+  "properties": {
+    "storykitPackageName": {
+      "title": "Specifies an NPM package that will provide the Storybook dependencies for the project.",
+      "description": "Storybook's conventional approach is for your app project to have direct dependencies\non NPM packages such as `@storybook/react` and `@storybook/addon-essentials`.  These packages have\nheavyweight dependencies such as Babel, Webpack, and the associated loaders and plugins needed to\nbuild the Storybook app (which is bundled completely independently from Heft).  Naively adding these\ndependencies to your app's package.json muddies the waters of two radically different toolchains,\nand is likely to lead to dependency conflicts, for example if Heft installs Webpack 5 but\n`@storybook/react` installs Webpack 4. To solve this problem, `heft-storybook-plugin` introduces the concept of a separate\n\"storykit package\".  All of your Storybook NPM packages are moved to be dependencies of the\nstorykit.  Storybook's browser API unfortunately isn't separated into dedicated NPM packages,\nbut instead is exported by the Node.js toolchain packages such as `@storybook/react`.  For\nan even cleaner separation the storykit package can simply reexport such APIs.",
```

Comment:
Trim out the `\n` chars. They aren't needed in the description

Reply:
Updated, BTW will the schema json generated to some docs?

Reply:
No these will not. I believe @iclanton investigated converting schemas to docs in the past, it's a non-trivial task. Usually schemas can provide VSCode tooltips when hovering over properties, however that's only when using the root level `$schema` field to link to the schema, which isn't possible in Heft plugin options. The documentation here is mostly for developer reference.

Reply:
You can also use it to validate the options object. This PR isn't doing that, but I figured that could just get integrated into the Heft vNext version.